### PR TITLE
first pass of all ideal changes

### PR DESCRIPTION
**Pull request by mejango**

### CODE REVIEW

File: contracts/JBChainlinkV3PriceFeed.sol [hunk 1]
```diff
@@ -41,7 +48,14 @@ contract JBChainlinkV3PriceFeed is IJBPriceFeed {
   */
   function currentPrice(uint256 _decimals) external view override returns (uint256) {
     // Get the latest round information. Only need the price is needed.
-    (, int256 _price, , , ) = feed.latestRoundData();
+    (uint80 roundId, int256 _price, , uint256 updatedAt, uint80 answeredInRound) = feed
+      .latestRoundData();
+    // Make sure the price isn't stale.
+    if (answeredInRound < roundId) revert STALE_PRICE();
```
Comment (drgorillamd) [inline contracts/JBChainlinkV3PriceFeed.sol:54]: Shouldn't we rather check the last timestamp? ETH/USD is updated for 0.5% deviation or 3600 seconds for instance (maybe some have longer heartbeat tho)

image

**Comment by mejango** [comment]
we should folds this in https://github.com/jbx-protocol/juice-contracts-v2/pull/279

**Comment by mejango** [comment]
and this https://github.com/jbx-protocol/juice-contracts-v2/pull/286

**Comment by mejango** [comment]
and this https://github.com/jbx-protocol/juice-contracts-v2/pull/288

File: contracts/JBSingleTokenPaymentTerminalStore.sol [hunk 2]
```diff
@@ -374,8 +374,11 @@ contract JBSingleTokenPaymentTerminalStore is IJBSingleTokenPaymentTerminalStore
       // A reference to the total amount that has been delegated.
       uint256 _totalDelegatedAmount;
 
+      // Keep a reference to the number of delegate allocations.
+      uint256 _numberOfDelegatedAmounts = delegateAllocations.length;
```
Comment (drgorillamd) [inline contracts/JBSingleTokenPaymentTerminalStore.sol:378]: Will revert this, stack is not happy

**Comment by mejango** [comment]
This PR was pulled out into its own V3 repo and merged to main there. it is not intended to be merged in this v2 repo.

**Pull request closed without merging**